### PR TITLE
provide dockstore version for download of upgrade jar

### PR DESCRIPTION
**Pull request by wshands**
Addresses https://ucsc-cgl.atlassian.net/browse/SEAB-3147

### CODE REVIEW

**Review by garyluu — APPROVED** [review]

**Review by denis-yuen — COMMENTED** [review]
Unclear from the linked ticket.
Why did this issue happen for 1.11 (and presumably not for previous releases)?
How did you decide between hotfix and 1.12 for this fix?

**Comment by wshands** [comment]
@denis-yuen RE: https://github.com/dockstore/dockstore-cli/pull/78#pullrequestreview-711710455
_Why did this issue happen for 1.11 (and presumably not for previous releases)?_
Code to download client jar at moment of upgrade was added September 14 https://github.com/dockstore/dockstore-cli/commit/4a9474a7eae4e5d38f317a102e57593510d002af, which was after  release 1.10.0 and newly incorporated into 1.11.0.
_How did you decide between hotfix and 1.12 for this fix?_ I didn't think of merging commits to the hotfix, but I can do that

File: dockstore-client/src/main/java/io/dockstore/client/cli/Client.java [hunk 1]
```diff
@@ -345,7 +347,15 @@ private static void downloadURL(String browserDownloadUrl, String installLocatio
             // Run the dockstore script with the 'self-install' argument so if it needs
             // to download the appropriate CLI JAR file it will do so now
             // and not the next time the user runs the dockstore script
-            Utilities.executeCommand(file.toPath().toString() + " self-install");
+            // Also set the Dockstore version environment variable to make
+            // sure that the dockstore script will download the new version of
+            // the client jar. The Dockstore version environment variable might
+            // be set to the previous version of Dockstore if the dockstore script
+            // was run to upgrade the Dockstore version.
+            Map<String, String> additionalEnvVarMap = new HashMap<>();
```
Comment (coverbeck) [inline dockstore-client/src/main/java/io/dockstore/client/cli/Client.java:355]: This is fine, but you could simplify it to `Map<String, String> additionalEnvVarMap = Collections.singletonMap("DOCKSTORE_VERSION", dockstoreVersion);` and get rid of the next line.

**Review by coverbeck — COMMENTED** [review]
LGTM, but I think you should target a hotfix branch.

File: dockstore-client/src/main/java/io/dockstore/client/cli/Client.java (hunk 1 above)
Comment (wshands) [reply dockstore-client/src/main/java/io/dockstore/client/cli/Client.java]: changed to incorporate your suggestion

**Comment by wshands** [comment]
Closing to create a new PR to merge to the new hotfix-1.11.1 branch

**Pull request closed without merging**